lib/weka/core/instances.rb

```diff
@@ -266,6 +307,50 @@ def map_attribute_type(type)
         return -1 unless Attribute::TYPES.include?(type.downcase.to_sym)
         Attribute.const_get(type.upcase)
       end
+
+      # Convert a hash whose keys are attribute names and values are attribute
+      #   values into an array containing attribute values in the order
+      #   of the Instances attributes.
+      #
+      # @param [Hash] attribute_values a hash whose keys are attribute
+      #   names and values are attribute values.
+      #
+      # @return [Array] an array containing attribute values in the
+      #   correct order
+      def attribute_values_from_hash(attribute_values)
+        names = attribute_names(true).map!(&:to_sym)
```

Comment:
Using  `map!` with a bang does not make to much sense, if we assign it afterwards anyway, right?
(Same in `attribute_values_to_hash()`)